[DOCS-6288] Add out-of-process ext docs to community, fix sys arch page, add repo concepts to 6.x and community (#303)
Alfresco/docs-alfresco: content-services/community/develop/sdk.md

| acs.debug.port | `number` | 8888 | The external port (i.e. outside container) for ACS Repository remote debugging.|
| postgres.port | `number` | 5555 | The external port (i.e. outside container) for PostgreSQL database.|

> There are some Alfresco Share related properties listed here, but they are not used unless you uncomment some code in the

Reviewer: Is this the terminology? 

```suggestion
There are some Alfresco Share related properties listed here, but they are not used unless you comment out some of the code in the
```
```suggestion
There are some Alfresco Share related properties listed here, but they are not used unless you uncomment some code in the
```

Author: It should be uncomment actually, not comment out

3. Click `Finish` to start importing the project into Eclipse.

   Before completing the import, Eclipse checks the completeness of the local Maven repository. If you already have a local repository that includes all the
   required dependencies, this task will finish relatively quickly. Otherwise, be patient and wait until the downloads are completed (it can take some time).

   Once the import is complete, a warning message may be displayed.

   ![sdk-dev-env-eclipse-warning]({% link content-services/images/sdk-dev-env-eclipse-warning.png %})

4. Click `Resolve All Later` to complete the import task.

5. Check the Markers tab in the bottom panel, where you may see some Maven problems. Expand the list and right click on a item with an error, then select `Quick Fix` and mark as shown.

   ![sdk-dev-env-eclipse-quickfix]({% link content-services/images/sdk-dev-env-eclipse-quickfix.png %})

6. Click `Finish` to confirm the fix.

   You may be asked to confirm your selection.

7. Repeat the fix for all similar issues you have. Note that these issues really depend on the archetype you used to generate the project.

>    Once done, you may see an error with description: Project configuration is not up-to-date with pom.xml.

Reviewer: ```suggestion
   Once done, you may see an error with description: Project configuration is not up-to-date with `pom.xml`.
```
```suggestion
   Once done, you may see an error with description: Project configuration is not up-to-date with pom.xml.
```

6. Give a descriptive name to the new server and set the `Server URL` as `http://localhost:8080/alfresco`.

   ![sdk-jrebel-eclipse-server]({% link content-services/images/sdk-jrebel-eclipse-server.png %})

7. Select the checkbox to enable the server synchronisation and the checkbox to `Synchronize on build`.

   ![sdk-jrebel-eclipse-servers]({% link content-services/images/sdk-jrebel-eclipse-servers.png %})

8. Select `Help > JRebel > Configuration > Projects` and select the checkbox to enable JRebel and the remote server support for required projects (the complete project in case of the Platform Jar archetype or the `PROJECT_ARTIFACT_ID-platform` and `PROJECT_ARTEFACTID-integration-tests` modules in case of the All-In-One archetype).

   ![sdk-jrebel-eclipse-projects]({% link content-services/images/sdk-jrebel-eclipse-projects.png %})

##### Installing JRebel using IntelliJ IDEA

1. Open IntelliJ and go to `Preferences > Plugins`….
2. Search for JRebel and select Install.
3. Restart IntelliJ to complete the installation.
4. Select `Preferences > JRebel > JRebel License` to activate your installation.

>    >**Note** that a license is required. In this step you will be able to request a trial license._

Reviewer: ```suggestion
   >**Note:** that a license is required. In this step you will be able to request a trial license._
```
```suggestion
   >**Note** that a license is required. In this step you will be able to request a trial license._
```
